tools/odbc/connection.cpp

```diff
@@ -14,12 +14,16 @@ using std::ptrdiff_t;
 SQLRETURN SQL_API SQLGetConnectAttr(SQLHDBC connection_handle, SQLINTEGER attribute, SQLPOINTER value_ptr,
                                     SQLINTEGER buffer_length, SQLINTEGER *string_length_ptr) {
 	duckdb::OdbcHandleDbc *dbc = nullptr;
-	if (ConvertConnection(connection_handle, dbc) != SQL_SUCCESS) {
-		return SQL_ERROR;
+	SQLRETURN ret = ConvertConnection(connection_handle, dbc);
+	if (ret != SQL_SUCCESS) {
+		return ret;
 	}
 
+	// TODO: If value_ptr is NULL shouldn't we return the length of the attribute, instead of returning SQL_ERROR?
+	// From the docs: "If ValuePtr is NULL, StringLengthPtr will still return the total number of bytes (excluding the
```

Comment:
This is actually very standard in the ODBC drive; many of these functions have the same signature.  Usually it is mentioned in the [diagnostics](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgetconnectattr-function?view=sql-server-ver16#diagnosticsl) if something should return an error (like if `value_ptr` is `NULL`), which in this case it doesn't, so I think it might be `SUCCESS_WITH_INFO`.

Also, the value returned in value_ptr isn't always a string.
But yes I agree that `string_length_ptr` should be set, but I think I'll do that in the next PR, there are still a lot of errors to work through.